ikra/soa/array_operator.h

```diff
@@ -0,0 +1,260 @@
+#ifndef SOA_ARRAY_OPERATOR_H
+#define SOA_ARRAY_OPERATOR_H
+
+#include "soa/array_field.h"
+#include "soa/util.h"
+
+#define TEMPLATE_ARRAY_FIELD \
+  template<typename T, size_t ArraySize, \
+           IndexType Capacity, uint32_t Offset, \
+           int AddressMode, int StorageMode, typename Owner>
+#define TEMPLATE_ARRAY_FIELD_W \
+  template<typename T, size_t ArraySize, \
+           IndexType Capacity1, uint32_t Offset1, \
+           int AddressMode1, int StorageMode1, typename Owner1, \
+           IndexType Capacity2, uint32_t Offset2, \
+           int AddressMode2, int StorageMode2, typename Owner2>
+#define AOS_ARRAY_FIELD(n) AosArrayField_<std::array<T, ArraySize>, \
+  Capacity ## n, Offset ## n, AddressMode ## n, StorageMode ## n, Owner ## n>
+#define SOA_ARRAY_FIELD(n) SoaArrayField_<T, ArraySize, \
+  Capacity ## n, Offset ## n, AddressMode ## n, StorageMode ## n, Owner ## n>
+#define STD_ARRAY std::array<T, ArraySize>
+
+#define IKRA_DEFINE_STD_ARRAY_OPERATORS_A(symbol) \
+  template<typename T, size_t N> \
+  std::array<T, N> operator symbol (std::array<T, N> left, \
```

Comment:
Provide overload where `left` and `right` are passed as reference. Put in separate commit/pull request. (Put TODO comment for now.)